File: languageSchoolProject/src/main/java/com/langschool/languageschoolproject/service/StudentService.java
package com.langschool.languageschoolproject.service;

import com.langschool.languageschoolproject.entity.Student;
import com.langschool.languageschoolproject.repo.StudentRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StudentService {

    @Autowired
    private StudentRepo repository;
    public Student saveStudent(Student student)
    {
        return repository.save(student);
    }

    public List<Student> saveStudents(List<Student> students)
    {
        return repository.saveAll(students);
    }

    public List<Student> getStudents()
    {
        return repository.findAll();
    }

    public Student getStudentById(int id)
    {
        return repository.findById(id).orElse(null);
    }
    public Student getStudentBySchoolIndex (String idx)
    {
        return repository.findBySchoolIndex(idx);

    }

    public String deleteStudent(int id)
    {
        repository.deleteById(id);
        return "Student with id "+id+" removed!";
    }

    public Student updateStudent(Student student)
    {
        Student existingStudent=repository.findById(student.getId()).orElse(null);
        existingStudent.setName(student.getName());
        existingStudent.setLastName(student.getLastName());
        existingStudent.setSchoolIndex(student.getSchoolIndex());
        return repository.save(existingStudent);
    }
}

File: react/my-app/src/services/StudentService.js
import axios from 'axios';

const TEACHER_API_BASE_URL = "http://localhost:9191/api";

class StudentService {

    getStudents(){
        return axios.get(TEACHER_API_BASE_URL + "/students");
    }

    createStudent(student){
        return axios.post(TEACHER_API_BASE_URL + "/addstudent" , student);
    }

    getStudentById(studentId){
        return axios.get(TEACHER_API_BASE_URL + '/studentById/' + studentId);
    }

    updateStudent(student){
        return axios.put(TEACHER_API_BASE_URL + '/updateStudent/', student);
    }

    deleteStudent(studentId){
        return axios.delete(TEACHER_API_BASE_URL + '/deleteStudent/' + studentId);
    }
}

export default new StudentService()